client/src/App.jsx

```diff
@@ -1,8 +1,11 @@
 import { useState, useEffect } from 'react'
 import './App.css'
+import Breadcrumbs from './components/breadcrumbs'
+import { BrowserRouter as Router, Route, Routes, useNavigate } from 'react-router-dom'
 
 function App() {
   const [files, setFiles] = useState(null)
+  const navigate = useNavigate()
```

Comment:
there is no need in useNavigate hook
why don't you try to work with files instead? 
update them and rerender breadcrambs. 
you pass handleBreadcrumbClick into breadcrumb component - use it there for changing `files` state